packages/wallet-adapter-core/src/utils/helpers.ts

```diff
@@ -97,10 +97,9 @@ export const getAptosConfig = (
     });
   }
 
-  return new AptosConfig({
-    network: Network.CUSTOM,
-    fullnode: networkInfo.url,
-  });
+  throw new Error(
+    "Invalid network, custom network not supported with Aptos wallet adapter to prevent user from using an unexpected network.",
```

Comment:
Maybe leave a comment telling the dev what they should do configuration-wise to avoid this issue.